<a href="https://colab.research.google.com/github/AjayNRG/DeepLearning/blob/master/1_MNIST_SampleDigits.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1. Loading the MNIST dataset in Keras

In [2]:
from keras.datasets import mnist

(train_images,train_labels),(test_images,test_labels) = mnist.load_data()

Using TensorFlow backend.


11493376/11490434 [==============================] - 0s 0us/step


In [3]:
train_images.shape

(60000, 28, 28)

In [4]:
len(train_labels)

60000

In [5]:
train_labels

array([5, 0, 4, ..., 5, 6, 8], dtype=uint8)

######And here’s the test data:

In [6]:
test_images.shape

(10000, 28, 28)

In [7]:
len(test_labels)

10000

In [8]:
test_labels

array([7, 2, 1, ..., 4, 5, 6], dtype=uint8)

# 2. Network Architecture

###### Let’s build the network—again, remember that you aren’t expected to understand everything about this example yet.

In [0]:
from keras import models
from keras import layers 

In [0]:
network = models.Sequential()
network.add(layers.Dense(512,activation='relu',input_shape=(28*28,)))
network.add(layers.Dense(10,activation='softmax'))

#3. The compilation step

In [0]:
network.compile(optimizer='rmsprop',loss='categorical_crossentropy',metrics=['accuracy'])

######Before training, we’ll preprocess the data by reshaping it into the shape the network expects and scaling it so that all values are in the [0, 1] interval. Previously, our training images, for instance, were stored in an array of shape (60000, 28, 28) of type uint8 with values in the [0, 255] interval. We transform it into a float32 array of shape (60000, 28 * 28) with values between 0 and 1.

#4. Preparing the image data

In [0]:
train_images =  train_images.reshape((60000,28*28))
train_images = train_images.astype('float32')/255

test_images = test_images.reshape((10000,28*28))
test_images = test_images.astype('float32')/255

#5. Preparing the labels

In [0]:
from keras.utils import to_categorical

train_labels = to_categorical(train_labels)
test_labels = to_categorical(test_labels)

######We’re now ready to train the network, which in Keras is done via a call to the network’s fit method—we fit the model to its training data:

In [16]:
network.fit(train_images,train_labels,epochs=5,batch_size=128)

Epoch 1/5
60000/60000 [==============================] - 5s 79us/step - loss: 0.2527 - accuracy: 0.9269
Epoch 2/5
60000/60000 [==============================] - 5s 76us/step - loss: 0.1015 - accuracy: 0.9700
Epoch 3/5
60000/60000 [==============================] - 4s 71us/step - loss: 0.0670 - accuracy: 0.9793
Epoch 4/5
60000/60000 [==============================] - 4s 69us/step - loss: 0.0487 - accuracy: 0.9851
Epoch 5/5
60000/60000 [==============================] - 4s 68us/step - loss: 0.0365 - accuracy: 0.9891


######We quickly reach an accuracy of 0.989 (98.9%) on the training data. Now let’s check that the model performs well on the test set, too:

In [17]:
test_loss, test_acc = network.evaluate(test_images,test_labels)
print(test_loss, test_acc)

10000/10000 [==============================] - 0s 45us/step
0.07762214210898383 0.9775000214576721
